pandas/core/groupby/groupby.py

```diff
@@ -1240,9 +1276,6 @@ def _python_agg_general(self, func, *args, **kwargs):
             assert result is not None
             key = base.OutputKey(label=name, position=idx)
 
-            if is_numeric_dtype(obj.dtype):
-                result = maybe_downcast_numeric(result, obj.dtype)
-
             if self.grouper._filter_empty_groups:
                 mask = counts.ravel() > 0
```

Comment:
@rhshadrach can i get your help in this nearby piece of code?  in all existing tests, when we get here, we have `mask.all()`.  trying to come up with a case where this doesnt hold (or prove that it must always hold).  any thoughts?

Reply:
Looks like this is now removed - guessing mask.all() always held.

Reply:
yep.

next thing to ask your help with :  IIRC you've done a lot of work in core.apply, which DataFrameGroupBy.aggregate uses.  id like to make SeriesGroupBy.aggregate and DataFrameGroupBy.aggregate share more code (or at least be more obviously-similar).  can i get your thoughts on how to achieve this (and whether its the effort)?

Reply:
Once we make aggregate always aggregate (PoC implemented in #40275), we can greatly simplify these methods. However, in order to do that we need to separate the apply/agg paths: currently apply uses agg for list/dicts and agg also uses apply for UDFs. I make a couple of attempts to do this but kept running into issues with changing behaviors without having a clear way to deprecate. This was the motivation for #41112. I plan to start working on that, assuming that's a good approach, in 1.3.